Documentation/package-dependency-flow.md

```diff
@@ -0,0 +1,140 @@
+# Package Dependency Flow
+
+This document describes how package dependencies are handled within source build.
+It described the mechanisms that exist to control which package versions are used.
+
+## Origins of Packages
+
+A source build must be self-contained, meaning the entire product must be built
+from source in an offline environment. This means all packages dependencies must
+be satisfied by one of the following:
+
+### Source-Build-Reference-Packages
+
+The first repo that is built as part of source build is
+[source-build-reference-packages](https://github.com/dotnet/source-build-reference-packages).
+This repo contains all of the reference packages used to build the product. The repo
+contains [tooling to generate new reference packages](https://github.com/dotnet/source-build-reference-packages?tab=readme-ov-file#adding-new-packages).
+
+### Current Source Built Packages
+
+This refers to all of the packages produced in the current build. The set of packages
+available to each repo varies based on its build order. If repo X has a dependency
+on package Y.1 but the repo Y hasn't been built yet, repo X cannot depend
+on the very latest version of the package.
```

Comment:
I think this would be better explained with concrete examples. Something like:

The msbuild repo can take a dependency on the current version of Microsoft.CodeAnalysis from roslyn because roslyn builds before msbuild. Conversely, since roslyn builds before msbuild, roslyn cannot take a dependency on the current version of Microsoft.Build; it can only take a dependency on the previously released version.

Reply:
Updated.